s3 error and autoscaling not starting

Hi, first thanks for project, I learned a lot from it.

I'm trying to start your kube cluster.

![image](https://cloud.githubusercontent.com/assets/9496645/15455912/81e3db82-2061-11e6-95a3-af89af31bb2d.png)

I have tried the deployment through a docker container first and now with an ubuntu vm. I have always the same issue. Do you have an idea on that s3 problem ? I'm admin with the aws account used.

### s3 cp

s3 cp issue is linked to aws configuration.

At first, my `aws cli` was configured to `eu-central-1` and the region for the tack deployment was `us-west-1`. I switched my `aws cli` configuration to `us-west-1` and that s3 cp problem gone.

Maybe we could add the region explicitly [here](https://github.com/kz8s/tack/blob/master/modules/s3/s3.tf#L12) as an ENV ?

### autoscaling

The autoscaling still timeout.

Here are the outputs:

```
module.worker.aws_autoscaling_group.worker: Still creating... (9m50s elapsed)
module.worker.aws_autoscaling_group.worker: Still creating... (10m0s elapsed)
Error applying plan:

1 error(s) occurred:

* aws_autoscaling_group.worker: "worker": Waiting up to 10m0s: Need at least 5 healthy instances in ASG, have 3

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
makefiles/terraform.mk:11: recipe for target 'apply' failed
make: *** [apply] Error 1
➜  tack git:(master) ✗ make output kubeconfig
make: *** No rule to make target 'output'.  Stop.
➜  tack git:(master) ✗ terraform output kubeconfig
The output variable requested could not be found in the state
file. If you recently added this to your configuration, be
sure to run `terraform apply`, since the state won't be updated
with new output variables until that command is run.
➜  tack git:(master) ✗ kubectl get nodes
NAME                                       STATUS    AGE
ip-10-0-10-10.us-west-1.compute.internal   Ready     36m
ip-10-0-10-9.us-west-1.compute.internal    Ready     36m
ip-10-0-11-86.us-west-1.compute.internal   Ready     36m
➜  tack git:(master) ✗ 
```

I will let the cluster run to see if it finally scale. If it's so long, maybe it shouldn't fail for that ? Or maybe that scaling is abnormally slow ?

### kubectl configuration

kubectl was configured this time.
Note that `terraform output kubeconfig` didn't work for me:

```
➜  tack git:(master) ✗ terraform output kubeconfig
The output variable requested could not be found in the state
file. If you recently added this to your configuration, be
sure to run `terraform apply`, since the state won't be updated
with new output variables until that command is run.
```

I suppose it's a problem linked to the fact terraform didn't finished sucessfully (due to the timeout).

--- a/modules.tf
+++ b/modules.tf
@@ -6,6 +6,7 @@ module "s3" {
   k8s-version = "${var.k8s.version}"
   name = "${ var.name }"
   podmaster-image = "${ var.k8s.podmaster-image }"
+  region = "${ var.aws.region }"
 }
 
 module "vpc" {

--- a/modules/s3/io.tf
+++ b/modules/s3/io.tf
@@ -3,5 +3,6 @@ variable "hyperkube-image" {}
 variable "k8s-version" {}
 variable "name" {}
 variable "podmaster-image" {}
+variable "region" {}
 
 output "bucket-prefix" { value = "${ var.bucket-prefix }" }

--- a/modules/s3/s3-cp
+++ b/modules/s3/s3-cp
@@ -39,7 +39,8 @@ MANIFESTS=manifests/system
 
 # cp tls assets
 echo "❤ Copy tls assets to s3 ${BUCKET}"
-aws s3 cp .cfssl s3://${BUCKET}/ssl --recursive --exclude "*" --include "*.tar"
+aws s3 cp .cfssl s3://${BUCKET}/ssl --recursive --exclude "*" --include "*.tar" \
+  --region ${REGION}
 echo "✓ tls copy success"
 
 # mk tmp dir for manifests and do substituions
@@ -52,5 +53,5 @@ sed -i.bak 's|${HYPERKUBE}|'"$HYPERKUBE|g" "$TMP/"*.yml
 echo "❤ Tar and copy etc manifests to ${BUCKET}"
 tar -cf - \
   -C $TMP kube-apiserver.yml kube-controller-manager.yml kube-proxy.yml kube-scheduler.yml \
-  | aws s3 cp - s3://${BUCKET}/manifests/etc.tar
+  | aws s3 cp - s3://${BUCKET}/manifests/etc.tar --region ${REGION}
 echo "✓ Tar and copy etc manifests success"

--- a/modules/s3/s3.tf
+++ b/modules/s3/s3.tf
@@ -2,14 +2,21 @@ resource "aws_s3_bucket" "ssl" {
   acl = "private"
   bucket = "${ var.bucket-prefix }"
   force_destroy = true
+
   tags {
+    builtWith = "terraform"
     Cluster = "${ var.name }"
     Name = "k8s"
-    builtWith = "terraform"
   }
 
   provisioner "local-exec" {
-    command = "HYPERKUBE=${ var.hyperkube-image } PODMASTER=${ var.podmaster-image } ${ path.module }/s3-cp ${ var.bucket-prefix }"
+    command = <<EOF
+      HYPERKUBE=${ var.hyperkube-image } \
+      PODMASTER=${ var.podmaster-image } \
+      REGION=${ var.region } \
+        ${ path.module }/s3-cp ${ var.bucket-prefix }
+EOF
   }
 
+  region = "${ var.region }"
 }

